### ISSUE TITLE
gzdoom crashes when exiting animation

### ISSUE THREAD

**Issue by x3003** [coelckers/gzdoom#1396]
With the latest git version of blade of agony. When starting a new game, the chapter doesn't matter, when pressing the use key during the first animation,  gzdoom crashes. Very fatal error. This behaviour was introducted in v4.7pre6, as pre4 was still good.

[CrashReport.zip](https://github.com/coelckers/gzdoom/files/6547859/CrashReport.zip)

**Comment by vsonnier** [comment]
Actually, I think I have the same problem with Brutal Doom Black Edition. (a not yet released version for Gzdoom 4.60),

The problem started with the commit 5f02b92cd0843026a6bc20743915333041c96cd7 "font system overhaul"

**Comment by vsonnier** [comment]
Using a debug version, I got an access violation at `font.cpp (673)`:

```cpp
//==========================================================================
//
// FFont :: GetColorTranslation
//
//==========================================================================

int FFont::GetColorTranslation (EColorRange range, PalEntry *color) const
{
	if (noTranslate)
	{
		PalEntry retcolor = PalEntry(255, 255, 255, 255);
		if (range >= 0 && range < NumTextColors && range != CR_UNTRANSLATED)
		{
			retcolor = TranslationColors[range];
			retcolor.a = 255;
		}
		if (color != nullptr) *color = retcolor;
	}
	if (range == CR_UNDEFINED)
		return -1;
	else if (range >= NumTextColors)
		range = CR_UNTRANSLATED;
	return Translations[range]; // <== Translation size is = 0 here, while range is CR_UNTRANSLATED (11)
}
```

Pre-packageed Black edition : https://drive.google.com/file/d/1n-ujxdwC54TH7mlsog5smpqYMnGYxCyM/view?usp=sharing

**Comment by vsonnier** [comment]
Removed the BlackEdition link, here is a minimal example demonstrating the problem : 
[Font_Crash_Use_Case.zip](https://github.com/coelckers/gzdoom/files/6551401/Font_Crash_Use_Case.zip)

**Comment by coelckers** [comment]
Duplicate: https://forum.zdoom.org/viewtopic.php?f=2&t=72426

**Comment by x3003** [comment]
Seems fixed.

**Comment by LordMisfit** [comment]
So it looks like this was the other crash issue I was having after the font changes in 4.7pre6 were added. 4.7pre12 and on seems to have fixed the problem. I was having trouble narrowing it down, though before I found the new version, I noticed it seemed to be crashed due to an ACS script, probably uses of HudMessage. I had also loaded a smaller mod that had ACS hud code, and it had also crashed the same, so I guess it was related to Untranslated palettes or so?
